### PR TITLE
chore(@aws-amplify/datastore): docstrings, small refactors, cleanup unused predicate function

### PR DESCRIPTION
**Pull request by svidgen**
#### Description of changes

A batch of refactors, comments, etc., intended to improve readability.

1. A handful of docstrings to help explain and add hover-text to new methods from lazy loading + nested predicates.
2. Handful of explanatory comments throughout model lazy loader
3. Small refactor to model setter/getter creation to a) remove unused vars, b) leverage `Relationship` class's `allFrom` method instead of duplicating that logic inline.
4. Small refactor to storage predicate builder, moving away from using predicate storage builder.

#### Description of how you validated changes

No functional changes intended. Tests are expected to pass.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/src/predicates/next.ts [hunk 1]
```diff
@@ -758,12 +758,6 @@ export class GroupCondition {
  * `predicateFor()` returns objecst with recursive getters. To facilitate this,
  * a `query` and `tail` can be provided to "accumulate" nested conditions.
  *
- * TODO: the sortof-immutable algorithm was originally done to support legacy style
- * predicate branching (`p => p.x.eq(value).y.eq(value)`). i'm not sure this is
- * necessary or beneficial at this point, since we decided that each field condition
- * must flly terminate a branch. is the strong mutation barrier between chain links
- * still necessary or helpful?
```
Comment (svidgen) [inline packages/datastore/src/predicates/next.ts:765]: Realistically, we'll never revisit this immutability. It has value if we ever need to extend predicates. And if we refactor here (which is possible), it'll be done holistically; not for the purpose of addressing this specific algorithmic choice.

File: packages/datastore/src/predicates/index.ts [hunk 2]
```diff
@@ -290,17 +317,39 @@ export class ModelPredicateCreator {
...
+	 * 	]
+	 * }
+	 * ```
+	 *
+	 * @param modelDefinition The model that the AST/predicate must be compatible with.
+	 * @param ast The graphQL style AST that should specify conditions for `modelDefinition`.
+	 */
 	static createFromAST(
 		modelDefinition: SchemaModel,
 		ast: any
```
Comment (svidgen) [inline packages/datastore/src/predicates/index.ts:355]: A separate PR will attempt to address the `any` here.

File: packages/datastore/src/predicates/index.ts [hunk 3]
```diff
@@ -231,53 +268,43 @@ export class ModelPredicateCreator {
...
+	 * 				{ favoriteFood: { eq: 'tacos' } },
+	 * 			]
+	 * 		}}
+	 * 	]
+	 * }
+	 * ```
+	 *
+	 * @param gql GraphQL style filter node.
+	 */
+	static transformGraphQLFilterNodeToPredicateAST(gql: any) {
```
Comment (svidgen) [inline packages/datastore/src/predicates/index.ts:298]: A separate PR will attempt to address the `any` here.

**Review by iartemiev — APPROVED** [review]

**Review by david-mcafee — APPROVED** [review]
This is fantastic! Thanks for doing this!

**Pull request merged**